pallets/migrations/src/lib.rs

```diff
@@ -31,13 +32,19 @@ pub mod pallet {
 
 	#[derive(Copy, Clone, PartialEq, Eq, Encode, Decode, RuntimeDebug)]
 	/// A Migration that must happen on-chain upon a runtime-upgrade
-	pub struct Migration {
+	pub trait Migration {
 		// TODO: this would involve some metadata about the migration as well as a means of calling
 		// the actual migration function
+
+		fn friendly_name() -> str;
 	}
 
 	/// Our list of migrations. Any ordering considerations can be specified here (?).
-	// const MIGRATIONS: // TODO: this would be a constant vec (or similar) of individual migrations
+	static MIGRATIONS: [Migration] = [
+		MM_001_AuthorMappingAddDeposit,
+		MM_002_StakingFixTotalBalance,
+		MM_003_StakingTransitionBoundedSet,
+	];
```

Comment:
Cool, I like this design.

Reply:
So the migration would not execute in this pallet, but execution would be tracked in this pallet? Where is each migration executed and how is that passed to this pallet?

Reply:
Yeah, I imagine the migration could would live inside the crate it belongs to. Perhaps each migration could be their own `mod` and have unit tests, etc

I'm envisioning having a function pointer or similar in the Migration trait -- some way to invoke the actual migration code.

Reply:
> function pointer or similar

https://doc.rust-lang.org/reference/types/trait-object.html I think a trait object and dynamic dispatch is what you want here.

I'm not sure about `static` but I know you could do something like 
```rust
let migrations: Vec<Box<dyn Migration>> = vec![
  Box::new(MM_001_AuthorMappingAddDeposit),
  Box::new(MM_002_StakingFixTotalBalance),
  Box::new(MM_003_StakingTransitionBoundedSet),
]
```